### PR TITLE
Grid change

### PR DESCRIPTION
**Pull request by wfajczyk**
# Description
Change query system to grid filter.

Issue # 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [ ] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/grid/src/QueryInterface.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Grid;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ergonode\Grid\Request\FilterValueCollection;
+
+interface QueryInterface
```
Comment (wfajczyk) [inline module/grid/src/QueryInterface.php:14]:
```suggestion
interface DbalDataSetQueryInterface 
```

File: module/grid/src/DbalDataSet.php [hunk 2]
```diff
@@ -66,4 +65,15 @@ public function countItems(FilterValueCollection $values, array $columns = []):
 
         return 0;
     }
+
+    /**
+     * @param ColumnInterface[] $columns
+     */
+    public function getQueryBuilder(FilterValueCollection $values, array $columns = []): QueryBuilder
```
Comment (piotrkreft) [inline module/grid/src/DbalDataSet.php:72]: It's redundant now I guess. Basically as other changes in this file

File: module/product/src/Infrastructure/Persistence/DataSet/DbalProductDataSet.php [hunk 3]
```diff
@@ -113,6 +101,22 @@ public function countItems(FilterValueCollection $values, array $columns = []):
         return 0;
     }
 
+    /**
+     * @param ColumnInterface[] $columns
+     */
+    public function getQueryBuilder(FilterValueCollection $values, array $columns = []): QueryBuilder
```
Comment (piotrkreft) [inline module/product/src/Infrastructure/Persistence/DataSet/DbalProductDataSet.php:107]: This should be moved to inherited implementation

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — APPROVED** [review]

File: module/product/src/Infrastructure/Persistence/DataSet/DbalProductDataSetQueryBuilder.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+declare(strict_types=1);
+
+namespace Ergonode\Product\Infrastructure\Persistence\DataSet;
+
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ergonode\Grid\ColumnInterface;
+use Ergonode\Grid\DbalDataSetQueryInterface;
+use Ergonode\Grid\Request\FilterValueCollection;
+
+class DbalProductDataSetQueryBuilder extends DbalProductDataSet implements DbalDataSetQueryInterface
```
Comment (rprzedzik) [inline module/product/src/Infrastructure/Persistence/DataSet/DbalProductDataSetQueryBuilder.php:16]:
```suggestion
class DbalQueryBuilderProductDataSet extends DbalProductDataSet implements DbalDataSetQueryInterface
```

**Review by wiewiurdp — APPROVED** [review]

File: module/grid/src/DataSet/DataSetGridId.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+declare(strict_types=1);
+
+namespace Ergonode\Grid\DataSet;
+
+use Doctrine\DBAL\FetchMode;
+use Ergonode\Grid\GridConfigurationInterface;
+use Ergonode\Grid\GridInterface;
+use Ergonode\Grid\DbalDataSetQueryInterface;
+
+class DataSetGridId
```
Comment (rprzedzik) [inline module/grid/src/DataSet/DataSetGridId.php:16]: This ins not a data set object is ProductIdsQuery or productIdsProvider ors something similar

File: module/grid/src/DataSet/DataSetGridId.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public function getItems(
```
Comment (rprzedzik) [inline module/grid/src/DataSet/DataSetGridId.php:18]:
```suggestion
    public function getProductIds(
```

File: module/grid/src/DataSet/DataSetGridId.php [hunk 7]
```diff
@@ -0,0 +1,31 @@
+<?php
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Grid\DataSet;
```
Comment (rprzedzik) [inline module/grid/src/DataSet/DataSetGridId.php:9]: This class use Grid andr DataSet but this is not any one of it. Sholud be in different namespace

**Pull request merged**